lib/ImageProcessor.js

```diff
@@ -85,13 +88,18 @@ class ImageProcessor {
             resize.bucket = resize.bucket || bucket;
             resize.jpegOptimizer = resize.jpegOptimizer || jpegOptimizer;
 
-            promise = promise.then(() => this.execResizeImage(resize, imageData).then((image) => {
-                this.fileSystem.putObject(image);
-            }));
-            processedImages++;
+            promise = promise
+              .then(() => this.execResizeImage(resize, imageData))
+              .then((image) => this.fileSystem.putObject(image))
+              .then(() => Promise.resolve(++processedImages));
         });
 
-        return promise.then(() => new Promise((resolve) => resolve(processedImages)));
+        if ( removeAfterSuccess ) {
```

Comment:
Ah, I understand what you mean. I thought `after completion`, but uploaded binary data will keep in this process as `imageData`, so we can delete uploaded object in `backup` section.
I'll fix that.